kafka-client/src/main/java/dev/responsive/kafka/api/KeyValueStoreOptions.java

```diff
@@ -0,0 +1,92 @@
+package dev.responsive.kafka.api;
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+/**
+ * Configuration options for building a Responsive {@link KeyValueStore}. Use {@link #named(String)}
+ * (or {@link #namedGlobal(String)} if this is a global store) to get a new object and pass that in
+ * to one of the {@link ResponsiveStoreFactory} APIs, or apply any of the additional optional
+ * configs via the {@code #withXXX} methods below.
+ */
+public class KeyValueStoreOptions<K, V> {
+
+  // required configs
+  private final String name;
+  private final boolean isGlobal;
+
+  // optional configs
+  private boolean truncateChangelog = false;
+  private Long ttlMs = null;
+  private Serde<K> keySerde;
+  private Serde<V> valueSerde;
+
+  /**
```

Comment:
can we use the builder pattern here so that we know objects of this class are immutable once they're built?